akka-docs/src/main/paradox/scala/project/migration-guide-2.4.x-2.5.x.md

```diff
@@ -714,6 +714,8 @@ final ActorRef throttler =
     .run(materializer);
 ```
 
+Note that when using `Sink.actorRef` the `sender()` will not be the original `sender()` of the messages that is sent to the `throttler` in the above examples. To define a valid `sender()` you can use `Sink.foreach` instead of `Sink.actorRef`. Inside `foreach` you can use ordinary `tell` to send the messages to the `target`. The `sender` parameter to `tell` can be a stable field or extracted from the elements in the stream. However, it must not access the `sender()` method of the enclosing actor if any, since foreach is run by the stream.
```

Comment:
Hmm, I had to read that note five times to understand what it is about (also because I was missing the context). It's also a bit unsatisfactory that you cannot keep the original sender which would probably require a `Source.actorRef` that produces elements inside of an "envelope" that also contains the sender.

Here's an alternative try to explain the situation which may or may not be clearer ;)

-------

Note, that when using `Sink.actorRef` the sender of the original message sent to the `thottler` ActorRef will be lost and messages will arrive at the `target` with the sender set to `ActorRef.noSender`. Using this construct it is currently impossible to keep the original sender. Alternatively, you can manually specify a static sender by replacing `Sink.actorRef` with `Sink.foreach(msg => target.tell(msg, whateverSender))`. You could also calculate a sender by inspecting the `msg`. Be cautious not to use `target.tell(msg, sender())` inside of `Sink.foreach` because the result of `sender()` is undefined or will fail when executed from within a stream.

-------

```
Note, that when using `Sink.actorRef` the sender of the original message sent to the `thottler` ActorRef will be lost and messages will arrive at the `target` with the sender set to `ActorRef.noSender`. Using this construct it is currently impossible to keep the original sender. Alternatively, you can manually specify a static sender by replacing `Sink.actorRef` with `Sink.foreach(msg => target.tell(msg, whateverSender))`. You could also calculate a sender by inspecting the `msg`. Be cautious not to use `target.tell(msg, sender())` inside of `Sink.foreach` because the result of `sender()` is undefined or will fail when executed from within a stream.
```